Easydict/Swift/View/SettingView/Tabs/ServiceConfigurationView/TextEditorCell.swift

```diff
@@ -17,72 +17,148 @@ struct TextEditorCell: View {
+        footnote: LocalizedStringKey? = nil,
+        minHeight: CGFloat? = nil,
+        maxHeight: CGFloat? = nil,
+        height: CGFloat? = nil
     ) {
         self.titleKey = titleKey
         self.placeholder = placeholder
+        self.alignment = alignment
+        self.footnote = footnote
+        self.minHeight = minHeight
+        self.maxHeight = maxHeight
+        self.height = height
         _value = .init(storedValueKey)
     }
 
     // MARK: Internal
 
     let titleKey: LocalizedStringKey
     @Default var value: String
-    let placeholder: LocalizedStringKey
+    let placeholder: LocalizedStringKey?
+    let alignment: TextAlignment
+    let footnote: LocalizedStringKey?
+    let minHeight: CGFloat?
+    let maxHeight: CGFloat?
+    let height: CGFloat?
 
     var body: some View {
-        HStack(alignment: .center, spacing: 20) {
-            Text(titleKey)
-
-            TrailingTextEditorWithPlaceholder(text: $value, placeholder: placeholder)
-                .padding(.horizontal, 3)
-                .padding(.top, 5)
-                .padding(.bottom, 7)
-                .font(.body)
-                .lineSpacing(5)
-                .scrollContentBackground(.hidden) // Refer https://stackoverflow.com/a/62848618/8378840
-                .scrollIndicators(.hidden)
-                .background(Color.clear)
-                .clipShape(corner)
-                .overlay(alignment: .center, content: {
-                    corner.stroke(Color(NSColor.separatorColor), lineWidth: 1)
-                })
-                .frame(minHeight: 55, maxHeight: 200) // min height is two lines, for English placeholder.
+        VStack(alignment: .leading, spacing: 10) {
+            HStack(alignment: .center, spacing: 20) {
+                Text(titleKey)
+                textEditor
+            }
+            footnoteView
         }
         .padding(10)
     }
 
     // MARK: Private
 
-    private let corner = RoundedRectangle(cornerRadius: 5)
+    private var textEditor: some View {
+        TextEditorWithPlaceholder(text: $value, placeholder: placeholder, alignment: alignment)
+            .padding(.horizontal, 3)
+            .padding(.vertical, 5)
+            .font(.body)
+            .lineSpacing(5)
+            .scrollContentBackground(.hidden)
+            .scrollIndicators(.hidden)
+            .background(Color.clear)
+            .clipShape(RoundedRectangle(cornerRadius: 5))
+            .overlay(RoundedRectangle(cornerRadius: 5).stroke(Color(NSColor.separatorColor), lineWidth: 1))
+            .frame(minHeight: minHeight, maxHeight: maxHeight)
+            .frame(height: height)
+    }
+
+    @ViewBuilder
+    private var footnoteView: some View {
+        if let footnote = footnote {
+            Text(footnote)
+                .font(.footnote)
+                .foregroundStyle(.gray)
+                .fixedSize(horizontal: false, vertical: true)
+                .lineSpacing(3)
+                .textSelection(.enabled)
```

Comment:
This text view will stretch after selecting text

<img width="532" alt="Screenshot 2024-09-24 at 21 30 44" src="https://github.com/user-attachments/assets/6b69a126-4e73-443c-8e43-2ed5196b3763">

Reply:
It looks like this is caused by `.lineSpacing(3)`.

Reply:
OK, it's been fixed.

But why does lineSpacing cause this issue? If I want to add lineSpacing, how can I avoid it?

Reply:
I have no idea of the root cause, maybe it's a SwiftUI bug.
Same issue here: https://stackoverflow.com/questions/76113476/swiftui-bug-text-view-is-changing-line-spacing-when-selected